document/CoreFormat.mdk

```diff
@@ -1409,6 +1425,165 @@ A `annotatedCodeLocation` object may contain a property named `tags`, whose valu
+
+A `rule` object contains information that describes a rule.
+
+### Constraints
+
+Either the `shortDescription` property (&sect;[#rule-shortDescription])
+or the `fullDescription` property (&sect;[#rule-fullDescription])
+or both must be present.
+
+### `id` property { #rule-id }
+
+A `rule` object shall contain a property named `id` whose value is a string containing
+a stable, opaque identifier for the rule.
+
+~ IsoExample
+`"CA2101"`
+~
+
+### `name` property { #rule-name }
+
+A `rule` object may contain a property named `name` whose value is a string containing
+a rule identifier that is understandable to an end user.
+If `name` contains implementation details that change over time,
+a tool author might alter a rule's `name` (while leaving
+the stable `id` property unchanged).
+
+~IsoNote
+A rule `name` is suitable in contexts where a readable identifier is preferable and where the
+lack of stability is not a concern.
+~
+
+~ IsoExample
+`"SpecifyMarshalingForPInvokeStringArguments"`
+~
+
+### `shortDescription` property { #rule-shortDescription }
+
+A `rule` object may contain a property named `shortDescription` whose value is a string containing
+a concise description of the rule. The `shortDescription` property should be a single sentence that is understandable
+when displayed in user interface contexts where the available space is limited to a single line of text.
+
+~ IsoExample
+`"Specify marshaling for P/Invoke string arguments"`
+~
+
+### `fullDescription` property { #rule-fullDescription}
+
+An `rule` object should contain a property named `fullDescription` whose value is a string that describes the rule.
+
+The `fullDescription` property should, as far as possible, provide details sufficient to enable resolution of any problem indicated by the result. 
+
+The first sentence of the `fullDescription` property should provide a concise description of the rule,
+suitable for display in cases where available space is limited.
+Tools that construct `fullDescription` in this way need not provide a value for the `shortDescription` property.
+Tools that do not construct `fullDescription` in this way should provide a value for the `shortDescription` property,
+because otherwise, the initial portion of `fullDescription` that a viewer displays where available space is limited
+might not be understandable.
+
+### `options` property { #rule-options }
+
+A `rule` object may contain a property named `options`, whose value is a JSON object
+consisting of a set of name/value pairs with arbitrary names.
+The `options` objects shall describe the set of configurable options supported by the rule.
+
+The value within each name/value pair shall be a string, which may be the empty string.
+The value shall not be a JSON array or sub-object.
+
+### `messageFormats` property { #rule-messageFormats }
+
+A `rule` object may contain a property named `messageFormats`, whose value is a JSON object
+consisting of a set of name/value pairs with arbitrary names.
+
```

Comment:
The following two paragraphs, plus the example, are reworked now that we've specified a format string.